src/explorer.api/Controllers/ExploreController.cs

```diff
@@ -92,16 +101,45 @@ public IActionResult Result(System.Guid exploreId)
         [ProducesResponseType(StatusCodes.Status404NotFound)]
         public IActionResult OtherActions() => NotFound();
 
-        private static ColumnExplorer? CreateNumericColumnExplorer(AircloakType type, JsonApiClient apiClient, Models.ExploreParams data)
+        private ColumnExplorer? CreateColumnExplorer(AircloakType type, Models.ExploreParams data)
         {
-            return type switch
+            var resolver = new AircloakQueryResolver(apiClient, data.DataSourceName);
+
+            var components = type switch
             {
-                AircloakType.Integer => new IntegerColumnExplorer(apiClient, data),
-                AircloakType.Real => new RealColumnExplorer(apiClient, data),
-                AircloakType.Text => new TextColumnExplorer(apiClient, data),
-                AircloakType.Bool => new BoolColumnExplorer(apiClient, data),
-                _ => null,
+                AircloakType.Integer => new ExplorerImpl[]
+                {
+                    new IntegerColumnExplorer(resolver, data.TableName, data.ColumnName),
+                    new MinMaxExplorer(resolver, data.TableName, data.ColumnName),
+                },
+                AircloakType.Real => new ExplorerImpl[]
+                {
+                    new RealColumnExplorer(resolver, data.TableName, data.ColumnName),
+                    new MinMaxExplorer(resolver, data.TableName, data.ColumnName),
+                },
+                AircloakType.Text => new ExplorerImpl[]
+                {
+                    new TextColumnExplorer(resolver, data.TableName, data.ColumnName),
+                },
+                AircloakType.Bool => new ExplorerImpl[]
+                {
+                    new BoolColumnExplorer(resolver, data.TableName, data.ColumnName),
+                },
+                _ => System.Array.Empty<ExplorerImpl>(),
             };
+
+            if (components.Length == 0)
+            {
+                return null;
+            }
+
+            var explorer = new ColumnExplorer();
```

Comment:
Here I would just pass the `components` array to the `ColumnExplorer` constructor, and then keep the `Spawn` method as private in the `ColumnExplorer` and use it just internally.

Reply:
I initially thought it might be useful to register the ExplorerImpls dynamically, which is why the Spawn method was public. In the end it's not really necessary, and I agree doing as you suggest makes things cleaner.

Reply:
Commit 09820d92584840a082f2236cd805a571997a4c1f